devlib/target.py

```diff
@@ -2893,6 +2898,127 @@ def _resolve_paths(self):
             self.executables_directory = '/tmp/devlib-target/bin'
 
 
+class TargetRunner:
+    '''
+    Class for running targets.
+    '''
+
+    banner = None
+    boot_timeout = None
+    runner_cmd = None
+    target = None
+    connection_settings = None
+
+    def __init__(self):
+
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+        self.logger.info('runner_cmd: %s', self.runner_cmd)
+
+        try:
+            self.runner_process = get_subprocess(self.runner_cmd, shell=True)
+        except Exception as ex:
+            raise HostError(f'Error while running "{self.runner_cmd}": {ex}') from ex
+
+        atexit.register(self.terminate_runner)
+
+        self.wait_boot_complete()
+
+        self.target.connect(check_boot_completed=False)
```

Comment:
Yes, default is `True`. Nothing fails since previous line (`self.wait_boot_completed()`) already waits for boot completion.
Leaving `check_boot_completed` as `True` just causes an extra function call.
I'll revisit this after addressing https://github.com/ARM-software/devlib/pull/667#discussion_r1467785241.